### DISCUSSION TITLE
[Question] Specify static IPs on AWS IPI

### DISCUSSION THREAD

**Discussion by dimidevops** [okd-project/okd#1833] [General]
Not a bug but a question.

I am trying to install openshift cluster (v4.14) using installer IPI, on a private AWS VPC with its own subnet. I define the machine subnet, cidr, and subnet ID but I cannot seem to find any related information on how to specify each ec2 node IP manually. Instead I keep getting randomly assigned free IPs by the DHCP (this is normal intended function). I know that the documentation mentioned to use Kubernetes nmstates operator to later define the states of the nic but I need to set the IPs before installation.

Is there an option to define each compute/worker ec2 node IP inside the install-config.yaml ?

**Version**
openshift-install 4.14.0-0.okd-2023-12-01-225814
built from commit 82051a1c3d42e394be353bbe2eacf348ccb75970
release image quay.io/openshift/okd@sha256:4510c50834c07e71521a68c021320b5e44978245c8031a858f74328e54e89e1c
release architecture amd64

**How reproducible**

**Log bundle**

**Comment by vrutkovs** [comment]
I don't think there is an option for that in AWS. The best way forward is to request an RFE via support (iiuc even developer subscriptions can do that)

**Comment by dimidevops** [comment]
I think this would probably not pass a RFE because per the documentation it is stated to use https://github.com/nmstate/kubernetes-nmstate to configure networking for each individual node after they are setup. The sad thing is that this operator is not available out of the box in the latest OKD release. Idk if I should open a RFE so the IP/Gateway/Nameservers can be defined inside the installer-config.yaml then translated to terraform for all cloud providers in order to manually define the instance network part like you the existing region/zone/vpc/subnet etc...
